pkg/sdk/testint/procedures_integration_test.go

```diff
@@ -0,0 +1,536 @@
+			WithReturns(procedureReturns).
+			WithHandler("FileReader.execute").
+			WithPackages([]sdk.ProcedurePackageRequest{*procedurePackage}).
+			WithProcedureDefinition(sdk.String(as))
+		err := client.Procedures.CreateProcedureForScala(ctx, request)
+		require.NoError(t, err)
+		t.Cleanup(cleanupProcedureHandle(id, []sdk.DataType{sdk.DataTypeVARCHAR}))
+
+		procedures, err := client.Procedures.Show(ctx, sdk.NewShowProcedureRequest())
+		require.NoError(t, err)
+		require.GreaterOrEqual(t, len(procedures), 1)
+	})
+
+	t.Run("create procedure for Scala: returns table", func(t *testing.T) {
+		name := "filter_by_role"
+		id := sdk.NewSchemaObjectIdentifier(databaseTest.Name, schemaTest.Name, name)
+
+		as := `
+	import com.snowflake.snowpark.functions._
+	import com.snowflake.snowpark._
+	object Filter {
+		def filterByRole(session: Session, tableName: String, role: String): DataFrame = {
+			val table = session.table(tableName)
+			val filteredRows = table.filter(col("role") === role)
+			return filteredRows
+		}
+	}`
+		column1 := sdk.NewProcedureColumnRequest().WithColumnName("id").WithColumnDataType(sdk.DataTypeNumber)
+		column2 := sdk.NewProcedureColumnRequest().WithColumnName("name").WithColumnDataType(sdk.DataTypeVARCHAR)
+		column3 := sdk.NewProcedureColumnRequest().WithColumnName("role").WithColumnDataType(sdk.DataTypeVARCHAR)
+		procedureReturnsTable := sdk.NewProcedureReturnsTableRequest().WithColumns([]sdk.ProcedureColumnRequest{*column1, *column2, *column3})
+		procedureReturns := sdk.NewProcedureReturnsRequest().WithTable(procedureReturnsTable)
+		procedureArgument1 := sdk.NewProcedureArgumentRequest().WithArgName("table_name").WithArgDataType(sdk.DataTypeVARCHAR)
+		procedureArgument2 := sdk.NewProcedureArgumentRequest().WithArgName("role").WithArgDataType(sdk.DataTypeVARCHAR)
+		procedurePackage := sdk.NewProcedurePackageRequest().WithPackage("com.snowflake:snowpark:latest")
+		request := sdk.NewCreateProcedureForScalaProcedureRequest(id).
+			WithOrReplace(sdk.Bool(true)).
+			WithRuntimeVersion(sdk.String("2.12")).
+			WithArguments([]sdk.ProcedureArgumentRequest{*procedureArgument1, *procedureArgument2}).
+			WithReturns(procedureReturns).
+			WithHandler("Filter.filterByRole").
+			WithPackages([]sdk.ProcedurePackageRequest{*procedurePackage}).
+			WithProcedureDefinition(sdk.String(as))
+		err := client.Procedures.CreateProcedureForScala(ctx, request)
+		require.NoError(t, err)
+		t.Cleanup(cleanupProcedureHandle(id, []sdk.DataType{sdk.DataTypeVARCHAR, sdk.DataTypeVARCHAR}))
+
+		procedures, err := client.Procedures.Show(ctx, sdk.NewShowProcedureRequest())
+		require.NoError(t, err)
+		require.GreaterOrEqual(t, len(procedures), 1)
+	})
+
+	t.Run("create procedure for Python: returns result data type", func(t *testing.T) {
+		name := "joblib_multiprocessing_proc"
+		id := sdk.NewSchemaObjectIdentifier(databaseTest.Name, schemaTest.Name, name)
+
+		as := `
+import joblib
```

Comment:
@We should probably run those with go-snowflake exec and implement proper call function in the following pr ?